helper-cli/src/main/kotlin/utils/PathExcludeGenerator.kt

```diff
@@ -37,13 +37,15 @@ internal object PathExcludeGenerator {
      * Return path excludes which likely but not necessarily apply to a source tree containing all given [filePaths]
      * which must be relative to the root directory of the source tree.
      */
-    fun generatePathExcludes(filePaths: Collection<String>): Set<PathExclude> {
-        val files = filePaths.mapTo(mutableSetOf()) { File(it) }
-
-        return generateExcludesForDirectories(files)
-    }
+    fun generatePathExcludes(filePaths: Collection<String>): Set<PathExclude> =
+        generateExcludesForDirectories(filePaths)
 
-    private fun generateExcludesForDirectories(files: Set<File>): Set<PathExclude> {
+    /**
+     * Return path excludes matching entire directories which likely but not necessarily apply to a source tree
+     * containing all given [filePaths] which must be relative to the root directory of the source tree.
+     */
+    fun generateExcludesForDirectories(filePaths: Collection<String>): Set<PathExclude> {
```

Comment:
Unrelated, but I'm mentioning it anyway: While re-reading the function name, it occurred to me that "...ForDirectories" sounds as if it expects `filePath` to point to directories. But instead what's meant is that the generated *excludes* refer to directories. So maybe `generateDirectoryExcludes` would be a better function name? That would also nicely complement `generatePathExcludes` with "Path" being the umbrella term for "Directory" and "File".